### PR TITLE
Hardening/adding valgrind sleep

### PR DESCRIPTION
**Pull request by raconte**
Not sure if works properly .

It has been modified the following test:
-0000_ngsi10_subscriptions/subscription_sequence_ontimeinterval.test
-0350_onanychange_subscriptions/onanychange_omit_cond_values.test
-1308_subscription_cache/subCacheThrottlingAndExpirationAfterSubUpdate.test
-1308_subscription_cache/subCacheThrottlingAndExpirationAfterRefresh.test
-1697_more_than_one_bad_input/more_than_one_bad_input.test

### CODE REVIEW

File: test/functionalTest/cases/1308_subscription_cache/subCacheThrottlingAndExpirationAfterSubUpdate.test [hunk 1]
```diff
@@ -116,15 +116,19 @@ echo
 echo
 
 
-echo "05. Wait 5 secs so that the throttling of 4 secs is OK"
+echo "05. To avoid a race condition in the order of notifications"
 echo "======================================================"
```
Comment (kzangeli) [inline test/functionalTest/cases/1308_subscription_cache/subCacheThrottlingAndExpirationAfterSubUpdate.test]: Justify the '====='-line also, to the same length as the 'echo 05. ...'

File: test/functionalTest/cases/1308_subscription_cache/subCacheThrottlingAndExpirationAfterSubUpdate.test [hunk 2]
```diff
@@ -116,15 +116,19 @@ echo
...
+duration=PT5S
+if [ "$VALGRIND" == 1 ]
 then
-  sleep 5
-else
-  sleep 4
+  duration=PT6S
 fi
-echo
+echo
```
Comment (kzangeli) [inline test/functionalTest/cases/1308_subscription_cache/subCacheThrottlingAndExpirationAfterSubUpdate.test]: space at end-of-line

File: test/functionalTest/cases/1308_subscription_cache/subCacheThrottlingAndExpirationAfterSubUpdate.test [hunk 3]
```diff
@@ -116,15 +116,19 @@ echo
...
+
+# Subscribe for 5 seconds (6 if VALGRIND)
+duration=PT5S
+if [ "$VALGRIND" == 1 ]
 then
-  sleep 5
-else
-  sleep 4
+  duration=PT6S
 fi
```
Comment (kzangeli) [inline test/functionalTest/cases/1308_subscription_cache/subCacheThrottlingAndExpirationAfterSubUpdate.test]: You only sleep for half a second, if non-valgrind?
I would have expected:

    sleep 4
    valgrindSleep 1

However, I imagine you have tested it. If it works, then all is OK.
Remember, the functests **without* valgrind must work also, actually, that is much more important that that they're working under valgrind.

**Comment by fgalan** [comment]
LGTM (appart from the comments already done by @kzangeli )

**Comment by kzangeli** [comment]
LGTM

File: test/functionalTest/cases/1308_subscription_cache/subCacheThrottlingAndExpirationAfterSubUpdate.test (hunk 2 above)
Comment (raconte) [inline test/functionalTest/cases/1308_subscription_cache/subCacheThrottlingAndExpirationAfterSubUpdate.test]: Fixed in d362c7f

File: test/functionalTest/cases/1308_subscription_cache/subCacheThrottlingAndExpirationAfterSubUpdate.test (hunk 3 above)
Comment (raconte) [inline test/functionalTest/cases/1308_subscription_cache/subCacheThrottlingAndExpirationAfterSubUpdate.test]: Fixed in d362c7f

File: test/functionalTest/cases/1308_subscription_cache/subCacheThrottlingAndExpirationAfterSubUpdate.test (hunk 1 above)
Comment (raconte) [inline test/functionalTest/cases/1308_subscription_cache/subCacheThrottlingAndExpirationAfterSubUpdate.test]: Fixed in d362c7f

**Comment by kzangeli** [comment]
I just ran a 'make test', after refreshing with develop branch and I got two errors in functests:

* 1308_subscription_cache/subCacheThrottlingAndExpirationAfterSubUpdate.test
* 1697_more_than_one_bad_input/more_than_one_bad_input.test

Please make sure all functests work (in normal functest mode).
That is much more important than that they work under valgrind.

**Comment by raconte** [comment]
I have seen that the 1697_more_than_one_bad_input/more_than_one_bad_input.test is not included in the develop branch .

When I run this test in my computer works properly but there is an error because the .out file and the .regexpect differs is in the step 4 ;

The expected is this one:
 `Raising alarm BadInput 0.0.0.0: service '/v2/noService' not found
 Repeated BadInput 0.0.0.0: JSON parse error
 Repeated BadInput 0.0.0.0: bad verb for url '/v2/entities', method 'GETT'
 Repeated BadInput 0.0.0.0: {"error":"LengthRequired","description":"Zero/No Content-Length in PUT/POST/PATCH request"}`

And my exit is this one:
` Raising alarm BadInput 127.0.0.1: service '/v2/noService' not found
 Repeated BadInput 127.0.0.1: JSON parse error
 Repeated BadInput 127.0.0.1: bad verb for url '/v2/entities', method 'GETT'
 Repeated BadInput 127.0.0.1: {"error":"LengthRequired","description":"Zero/No Content-Length in PUT/POST/PATCH request"}`

Is my local l0.

File: test/functionalTest/cases/1308_subscription_cache/subCacheThrottlingAndExpirationAfterSubUpdate.test [hunk 4]
```diff
@@ -116,8 +117,8 @@ echo
 echo
 
 
-echo "05. Wait 5 secs so that the throttling of 4 secs is OK"
-echo "======================================================"
+# 05. Wait 5 secs so that the throttling of 4 secs is OK
```
Comment (fgalan) [inline test/functionalTest/cases/1308_subscription_cache/subCacheThrottlingAndExpirationAfterSubUpdate.test]: "echo" seems to be missing...

Comment (raconte) [inline test/functionalTest/cases/1308_subscription_cache/subCacheThrottlingAndExpirationAfterSubUpdate.test]: Fixed in 84f5e10

**Pull request merged**